src/data/p5.TypedDict.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * @module Data 
+ * @submodule Dictionary
+ * @for p5
+ *@requires core
+ */
+
+var p5 = require('../core/core');
+
+//TESTED
+p5.prototype.createStringDict = function(key, value) {
+  return new p5.StringDict(key, value);
+}
+
+p5.TypedDict = function() {
+};
+
+//TESTED
+p5.TypedDict.prototype.size = function(){
+  return this.count;
+};
+
+//TESTED
+p5.TypedDict.prototype.get = function(key) {
+  if(this.data.hasOwnProperty(key)){
+    return this.data[key];
+  }else{
+    console.log(key + ' does not exist in this Dictionary');
+  }
+};
+
+//TESTED
+p5.TypedDict.prototype.set = function(key, value) {
+   if (arguments.length === 2) {
+    if(!this.data.hasOwnProperty(key)){
+      this.create(key, value);
+    } else {
+      this.data[key] = value;
+    }
+  }
+};
+
+//TESTED
+p5.TypedDict.prototype._add = function(obj) {
+ for (var key in obj) {
+   if(this._validate(obj[key])) {
+     this.data[key] = obj[key];
+     this.count++;
+   }
+  }
+}
+
+
+//TESTED
+p5.TypedDict.prototype.create = function(key, value) {
```

Comment:
it this a public method? seems like it could be private in which case let's call it _create.